sympy/physics/mechanics/functions.py

```diff
@@ -422,6 +447,11 @@ def find_dynamicsymbols(expression, exclude=None):
             raise TypeError("exclude kwarg must be iterable")
     else:
         exclude_set = set()
+    if isinstance(expression, Vector) and reference_frame is None :
```

Comment:
change this to:
```python
if isinstance(expresssion, Vector):
    if reference_frame is None:
        raise ValueError(" ... ")
    else:
        expression = expression.to_matrix(reference_frame)
```

Also I would expect a test case for this as well.

Reply:
I have made the requested change. A test case for this is already present as:

 # Test finding all dynamicsymbols in a vector with a given reference frame
    d, e, f = dynamicsymbols('d, e, f')
    A = ReferenceFrame('A')
    v = d * A.x + e * A.y + f * A.z
    sol = {d, e, f}
    assert find_dynamicsymbols(v, reference_frame=A) == sol